memory/deployment.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: memory
+  labels:
+    app: memory
+    hail.is/sha: "{{ code.sha }}"
+spec:
+  selector:
+    matchLabels:
+      app: memory
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: memory
+        hail.is/sha: "{{ code.sha }}"
+    spec:
+      serviceAccountName: memory
+{% if deploy %}
+      priorityClassName: production
+{% endif %}
+      containers:
+        - name: memory
```

Comment:
Not sure what you mean by this. If I have a redis instance running in a standalone instance, I'll need to use redis's auth stuff to ensure that only the memory container is capable of talking to the redis instance running in the other container within the k8s cluster. Seemed easier and more secure to run them in the same container, expose networking the same way we do for other services, and use a unix socket for communication.

If you mean that I should have some sort of container to manage load, etc. when we start sharding data, and that that should be the container called "memory" instead of the containers actually storing data, I can rename this container? It didn't seems super worth splitting them out right now since there's only one node.

Reply:
Pod is the natural security boundary in k8s.  Pods are a collection of containers.  Notice, in the YAML, containers is a list, not an item.  Containers in a pod are designed to work together.  They share a network and IP.  You can give them a shared file system.  I'm not sure how much of the container security reading group you attended, but containers are a collection of techniques for managing isolation **and** sharing of resources in running processes.  So while the containers in a pod are containers, they are less isolated than containers in two different pods, say.  The general recommendation is that each process in a pod should have its own container.  To give them a shared filesystem for the socket, use an emptyDir volume:

https://kubernetes.io/docs/concepts/storage/volumes/#emptydir

For more, see:

https://kubernetes.io/docs/concepts/workloads/pods/

"A Pod can encapsulate an application composed of multiple co-located containers that are tightly coupled and need to share resources."

Reply:
ah, cool! That also solves the "build redis from source" problem since there's a prebuilt redis container.